TidepoolOnboardingKitUI/Views/OnboardingTabView.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  OnboardingTabView.swift
+//  TidepoolOnboardingKitUI
+//
+//  Created by Darin Krauss on 1/28/21.
+//  Copyright © 2021 Tidepool Project. All rights reserved.
+//
+
+import SwiftUI
+
+struct OnboardingTabView: View {
+    @Environment(\.complete) var complete
+
+    @State private var selectedTabIndex = 0
+
+    var tabs: [AnyView]
+
+    var body: some View {
+        ZStack {
+            Color(.systemBackground)
+                .edgesIgnoringSafeArea(.all)
+            GeometryReader { geometry in
+                TabView(selection: $selectedTabIndex) {
+                    ForEach(tabs.indices) { tabIndex in
+                        ScrollView {
+                            VStack {
+                                tabs[tabIndex]
+                                Spacer()
+                                pager(for: tabIndex)
+                                    .padding(.vertical)
+                                button(for: tabIndex)
+                            }
+                            .padding()
+                            .frame(minHeight: geometry.size.height)
+                        }
+                        .tag(tabIndex)
+                        .navigationBarHidden(true)
+                    }
+                }
+                .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
+            }
+        }
+    }
+
+    private func pager(for index: Int) -> some View {
+        HStack() {
+            ForEach(tabs.indices) { tabIndex in
+                Circle()
+                    .frame(width: 8, height: 8)
+                    .foregroundColor(tabIndex == index ? .accentColor : .gray)
+            }
+        }
+    }
```

Comment:
Is this because SwiftUI doesn't have a page control? It seems the UX is different than the typical page control (tapping on the side navigates backwards/forwards). Will this be confusing?

Reply:
SwiftUI does have a page control. It is an overlay on top of the TabView sub-views. In order to be able to use the built-in page control it would be necessary to fix the button to the bottom of the screen, add the page control above that and have the scrollable area (required because the text on the last Welcome view does not fit all on screen on smaller devices) be above the page control. I discussed the pros & cons of using the built-in page control and (due to the need for the scrollable area and to have that scrollable area be consistent with the rest of the onboarding workflow) have chosen the custom page control instead.